src/Watcher/IndexDocumentBuilderFactory.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\tao\elasticsearch\Watcher;
+
+use oat\tao\model\search\index\AbstractDocumentBuilderFactory;
+use oat\tao\model\search\index\IndexDocumentBuilderInterface;
+
+class IndexDocumentBuilderFactory extends AbstractDocumentBuilderFactory
+{
+    const AVAILABLE_DOCUMENT_BUILDERS = [
+        'http://www.tao.lu/Ontologies/TAOItem.rdf#Item' => 'oat\\tao\\elasticsearch\\Watcher\\Resources\\ItemIndexDocumentBuilder',
+        'http://www.tao.lu/Ontologies/TAOTest.rdf#Test' => 'oat\\tao\\elasticsearch\\Watcher\\Resources\\GenericIndexDocumentBuilder',
+        'http://www.tao.lu/Ontologies/TAOGroup.rdf#Group' => 'oat\\tao\\elasticsearch\\Watcher\\Resources\\GenericIndexDocumentBuilder',
+        'http://www.tao.lu/Ontologies/TAODelivery.rdf#Delivery' => 'oat\\tao\\elasticsearch\\Watcher\\Resources\\GenericIndexDocumentBuilder',
+        'http://www.tao.lu/Ontologies/TAOResult.rdf#DeliveryResult' => 'oat\\tao\\elasticsearch\\Watcher\\Resources\\GenericIndexDocumentBuilder',
+        'http://www.tao.lu/Ontologies/TAOSubject.rdf#Subject' => 'oat\\tao\\elasticsearch\\Watcher\\Resources\\TesttakerIndexDocumentBuilder',
+        'unclassified' => 'oat\\tao\\elasticsearch\\Watcher\\Resources\\UnclassifiedIndexDocumentBuilder'
```

Comment:
Maybe this one could have its own constant?

I ask this because first we check if the resourceType is present on `AVAILABLE_DOCUMENT_BUILDERS` and if is not we use one from the same Array

Reply:
Yes and no :) 
My concern is that moving this "unclassified" builder to own constant may add additional questions, like: "we have available document builders array, so why one of the available builder is out of it?"